phpunit/class-wp-theme-json-test.php

```diff
@@ -2136,4 +2136,111 @@ public function test_internal_syntax_is_converted_to_css_variables() {
 		$this->assertEquals( 'var(--wp--preset--color--s)', $styles['blocks']['core/quote']['variations']['plain']['color']['background'], 'Style variations: Assert the internal variables are convert to CSS custom variables.' );
 
 	}
+
+	public function test_get_global_styles_values() {
+		$primary_color   = '#9DFF20';
+		$secondary_color = '#9DFF21';
+		$contrast_color  = '#000';
+		$large_font      = '18px';
+		$small_font      = '12px';
+		$theme_json      = new WP_Theme_JSON_Gutenberg(
+			array(
+				'version'  => WP_Theme_JSON_Gutenberg::LATEST_SCHEMA,
+				'settings' => array(
+					'color'      => array(
+						'palette' => array(
+							'theme' => array(
+								array(
+									'color' => $primary_color,
+									'name'  => 'Primary',
+									'slug'  => 'primary',
+								),
+								array(
+									'color' => $secondary_color,
+									'name'  => 'Secondary',
+									'slug'  => 'secondary',
+								),
+								array(
+									'color' => $contrast_color,
+									'name'  => 'Contrast',
+									'slug'  => 'contrast',
+								),
+							),
+						),
+					),
+					'typography' => array(
+						'fontSizes' => array(
+							array(
+								'size' => $small_font,
+								'name' => 'Font size small',
+								'slug' => 'small',
+							),
+							array(
+								'size' => $large_font,
+								'name' => 'Font size large',
+								'slug' => 'large',
+							),
+						),
+					),
+				),
+				'styles'   => array(
+					'elements' => array(
+						'button' => array(
+							'color'      => array(
+								'text' => 'var(--wp--preset--color--contrast)',
+							),
+							'typography' => array(
+								'fontSize' => 'var(--wp--preset--font-size--small)',
+							),
+						),
+					),
+					'blocks'   => array(
+						'core/post-terms' => array(
+							'typography' => array( 'fontSize' => 'var(--wp--preset--font-size--small)' ),
+							'color'      => array( 'background' => 'var(--wp--preset--color--secondary)' ),
+						),
+						'core/navigation' => array(
+							'elements' => array(
+								'link' => array(
+									'color'      => array(
+										'background' => 'var(--wp--preset--color--primary)',
+										'text'       => 'var(--wp--preset--color--secondary)',
+									),
+									'typography' => array(
+										'fontSize' => 'var(--wp--preset--font-size--large)',
+									),
+								),
+							),
+						),
+						'core/quote'      => array(
+							'typography' => array( 'fontSize' => 'var(--wp--preset--font-size--large)' ),
+							'color'      => array( 'background' => 'var(--wp--preset--color--primary)' ),
+							'variations' => array(
+								'plain' => array(
+									'typography' => array( 'fontSize' => 'var(--wp--preset--font-size--small)' ),
+									'color'      => array( 'background' => 'var(--wp--preset--color--secondary)' ),
+								),
+							),
+						),
+					),
+				),
+			)
+		);
+
+			$styles = $theme_json->get_styles_with_values();
```

Comment:
My editor (VSCode) reports that something is off with the indentation. I opened the file with the `less` command to make sure it wasn't my editor, and I still see how the indentation is off.

![image](https://github.com/WordPress/gutenberg/assets/583546/50089f93-f720-4600-9408-831e7d6d2163)

I wonder why the lint job is not reporting it :thinking: I also run `npm run format:php` (which is supposed to fix this kind of things) and it didn't find any fixable errors.

Reply:
I'm not sure what happens, so I'm going to summon people who know more than I do on this topic and have worked in this area recently: @ObliviousHarmony @noahtallen @anton-vlasenko Would you know why this linting issue is not reported in CI? I cannot get it to fail locally, either.

I changed some of the tabs to spaces, and it fails (as it should). Then I created this piece of art (which is a linting issue) and `npm run lint:php` still passes:

![image](https://github.com/WordPress/gutenberg/assets/583546/63a59507-d224-43b7-a956-300f2bdfc67a)

Reply:
@samnajian in the meantime, would you mind fixing this manually?

Reply:
Actually, `npm run lint:php` fails due to some permissions:

![image](https://github.com/WordPress/gutenberg/assets/583546/1ee5c86f-11ff-46a6-af00-dc1b7d07307b)

Reply:
I couldn't replicate the permissions issue, though it could probably be resolved by re-creating the container with wp-env (we recently updated its permissions model to ideally work better in most cases).

However, you can execute `composer run-script lint` locally without needing wp-env. (If you have the composer binary installed yourself, that is.) On this branch, that doesn't give me any errors either. Given that CI isn't reporting errors either, there must be a difference between Gutenberg's config and whatever your editor is using... it seems like Gutenberg's phpcs config may not be handling indentation properly? That's outside my realm of expertise though 😅

Reply:
@oandregal should be fixed in 0781b38438ee4cde66dc53e6dec3f16c9e7b20e6